aws/resource_aws_wafregional_web_acl.go

```diff
@@ -0,0 +1,231 @@
+		Read:   resourceAwsWafRegionalWebAclRead,
+		Update: resourceAwsWafRegionalWebAclUpdate,
+		Delete: resourceAwsWafRegionalWebAclDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"default_action": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"metric_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rules": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": &schema.Schema{
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": &schema.Schema{
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"priority": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"rule_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafRegionalWebAclCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+	region := meta.(*AWSClient).region
+
+	wr := newWafRegionalRetryer(conn, region)
+	out, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		params := &waf.CreateWebACLInput{
+			ChangeToken:   token,
+			DefaultAction: expandDefaultAction(d),
+			MetricName:    aws.String(d.Get("metric_name").(string)),
+			Name:          aws.String(d.Get("name").(string)),
+		}
+
+		return conn.CreateWebACL(params)
+	})
+	if err != nil {
+		return err
+	}
+	resp := out.(*waf.CreateWebACLOutput)
+	d.SetId(*resp.WebACL.WebACLId)
+	return resourceAwsWafRegionalWebAclUpdate(d, meta)
+}
+
+func resourceAwsWafRegionalWebAclRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+	params := &waf.GetWebACLInput{
+		WebACLId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetWebACL(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "WAFNonexistentItemException" {
+			log.Printf("[WARN] WAF Regional ACL (%s) not found, error code (404)", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	defaultAction := flattenDefaultActionWR(resp.WebACL.DefaultAction)
+	if defaultAction != nil {
+		if err := d.Set("default_action", defaultAction); err != nil {
+			return fmt.Errorf("error setting default_action: %s", err)
+		}
+	}
+	d.Set("name", resp.WebACL.Name)
+	d.Set("metric_name", resp.WebACL.MetricName)
+
```

Comment:
Do you mind setting `rules`/`rule` field here too? It'll most likely require creation of a flattener too, i.e.

```go
d.Set("rule", flattenWafRules(resp.WebACL.Rules))
```